File: marker_example/scripts/marker_example/tf_broadcaster.py
#! /usr/bin/env python3
import math
import rospy
import tf2_ros
from geometry_msgs.msg import TransformStamped
from dynamic_reconfigure.server import Server
from marker_example.cfg import MarkerExampleConfig


class TfBroadcaster:
    def __init__(self):
        self.broadcaster = tf2_ros.TransformBroadcaster()
        self.timer = rospy.Timer(rospy.Duration(0.02), self.timer_callback)
        self.transform_msg = TransformStamped()
        self.transform_msg.header.frame_id = 'map'
        self.transform_msg.child_frame_id = 'frame1'

        self.reconfig_srv = Server(MarkerExampleConfig, self.reconfig)
    
    def reconfig(self, config, level):
        self.transform_msg.transform.translation.x = config.x
        self.transform_msg.transform.translation.y = config.y
        self.transform_msg.transform.translation.z = 0

        self.transform_msg.transform.rotation.w = math.cos(0.5 * config.yaw)
        self.transform_msg.transform.rotation.z = math.sin(0.5 * config.yaw)
        return config

    def timer_callback(self, event):
        self.transform_msg.header.stamp = event.current_real
        self.broadcaster.sendTransform(self.transform_msg)


if __name__ == '__main__':
    rospy.init_node('tf_broadcaster')
    node_instance = TfBroadcaster()
    rospy.spin()

File: marker_example/src/tf_broadcaster.cpp
#include <ros/ros.h>
#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
#include <tf2_ros/transform_broadcaster.h>
#include <dynamic_reconfigure/server.h>
#include <marker_example/MarkerExampleConfig.h>

geometry_msgs::TransformStamped transform_msg;
std::shared_ptr<tf2_ros::TransformBroadcaster> broadcaster;

void timerCallback(const ros::TimerEvent& event)
{
  transform_msg.header.stamp = event.current_real;
  broadcaster->sendTransform(transform_msg);
}

void reconfig(marker_example::MarkerExampleConfig& config, uint32_t level)
{
  transform_msg.transform.translation.x = config.x;
  transform_msg.transform.translation.y = config.y;
  transform_msg.transform.translation.z = 0.0;

  tf2::Quaternion q;
  q.setRPY(0, 0, config.yaw);
  tf2::convert(q, transform_msg.transform.rotation);
}

int main(int argc, char** argv)
{
  ros::init(argc, argv, "tf_broadcaster");
  ros::NodeHandle node;

  broadcaster = std::make_shared<tf2_ros::TransformBroadcaster>();
  transform_msg.header.frame_id = "map";
  transform_msg.child_frame_id = "frame1";

  ros::Timer timer = node.createTimer(ros::Duration(0.02), timerCallback);

  dynamic_reconfigure::Server<marker_example::MarkerExampleConfig> srv;
  srv.setCallback(boost::bind(reconfig, _1, _2));

  ros::spin();
}